amc/pkg/istio/rest/dequeue_nodes.go

```
@@ -77,6 +91,73 @@ func DeQueueNodes(key string) {
 
 }
 
+func processSSLKeyPkiProfile(namespace string, secretName string, avimodel *nodes.AviObjectGraph, cache *utils.AviObjCache) bool {
+	var rest_ops []*utils.RestOp
+	utils.AviLog.Info.Printf("Found a ISTIO_MUTUAL secret to process for namespace: %s", namespace)
+	// First fetch the cache
+	sslkeycachekey := utils.NamespaceName{Name: secretName, Namespace: namespace}
+	sslkey, found := cache.SSLKeyCache.AviCacheGet(sslkeycachekey)
+	if found {
+		sslkey_obj, ok := sslkey.(*utils.AviSSLCache)
+		if !ok {
+			utils.AviLog.Warning.Printf("Invalid SSL Key object found. Cannot cast. Not doing anything")
+			return false
+		}
+		if avimodel == nil {
+			// DELETE request
+			rest_op := AviSSLKeyCertDel(sslkey_obj.Uuid, namespace)
+			rest_ops = append(rest_ops, rest_op)
+		}
+		// Proces it as a PUT request
+		sslnodes := avimodel.GetAviSSLCertKeys()
```

What if avimodel is `nil` here? This should throw an error, right?

---

Done.